### PR TITLE
[WAYP-491] website: Add runner diagram

### PR DESCRIPTION
**Pull request by amyrlam**
Summary:
- Add deployment diagram and some brief copy
- Fix typo along the way

Thinking we can add this in as an improvement, and continue to work on the larger improvements as specified in the parent ticket WAYP-451 / the related Google Doc for HCP Waypoint Cloud Docs. But will defer to what @ryenotbread / team thinks!

IS (can go to https://waypoint-ne3pnxb6n-hashicorp.vercel.app/waypoint/docs/runner):

image

image

~Note: May need rebase with https://github.com/hashicorp/waypoint/pull/4050, if 4050 gets merged~ done

### CODE REVIEW

**Comment by amyrlam** [comment]
Question:
- Should images be resized in their file since seems like they should all be full-width?
- Same as https://github.com/hashicorp/waypoint/pull/3998#issuecomment-1269056693 re: Vercel - waypoint failure

File: .changelog/3999.txt [hunk 1]
```diff
@@ -0,0 +1,3 @@
+```release-note:improvement
```
Comment (briancain) [inline .changelog/3999.txt:1]: 👋🏻 just a small note - no need for a changelog for website updates by the way

Comment (amyrlam) [reply .changelog/3999.txt]: thanks!

**Comment by ryenotbread** [comment]
Thanks so much for taking point on this Amy ✨ 

I wonder if the header should be "Runner architecture" or "Runner location" since this particular diagram shows how the remote runners communicate with the server? (Or we could swap it out with the ["deployment" diagram](https://drive.google.com/file/d/1Z_aSZrzDdMSamg_4YxqPTeKLZ4w81MnV/view?usp=sharing))

This might qualify for a separate PR, but I was also thinking the diagram that shows all the [runner types](https://drive.google.com/file/d/1Hc20BR6kSfLIBy8lDEcNU1KqPeP262S3/view?usp=sharing) could possibly live at the top of the page to illustrate the types of runners and where they live 👀

**Comment by amyrlam** [comment]
Thanks, this is good feedback. Because CLI runners is mentioned below, kind of rearranged them...let me know if you think it makes sense / can discuss in real time on Slack!

Will ask about the `website-mdx` failure

Also looking for a code review

**Comment by amyrlam** [comment]
Tried this for `website-mdx` error:

```
waypoint (main) 🌖 ~ go version
go version go1.16.7 darwin/amd64
waypoint (main) 🌖 ~ make gen/website-mdx
go run ./cmd/waypoint docs -website-mdx
../../../go/pkg/mod/github.com/hashicorp/vault/sdk@v0.6.0/helper/pluginutil/run_config.go:13:2: //go:build comment without // +build comment
make: *** [gen/website-mdx] Error 1
```

Could be needing Go 1.17+ from https://www.reddit.com/r/golang/comments/rn25ov/encountering_gobuild_comment_without_build/.

Followed https://gist.github.com/nikhita/432436d570b89cab172dcf2894465753 to install the latest, but my $PATH is different, so will ask on Slack tmrw. Am not using `asdf` or anything similar, had issues with it before and didn't seem necessary.

File: website/content/docs/runner/index.mdx [hunk 2]
```diff
@@ -18,6 +18,8 @@ server. This is important because runners are designed to be run in the same
 environment that your application is deployed into because the plugins such as Deploy
 need to talk directly to your deployment platform APIs.
 
+
+
```
Comment (BRKalow) [inline website/content/docs/runner/index.mdx:22]:
```suggestion
```

**Review by sabrinako — APPROVED** [review]
LGTM!

**Comment by amyrlam** [comment]
@ryenotbread gave the 👍 on Slack

Added @tunzor to take a look, following the Slack thread with Bryce

**Review by gregone — APPROVED** [review]

**Review by tunzor — APPROVED** [review]
LGTM!

**Pull request merged**